### PR TITLE
Fix drawing of 90 degree args

### PR DESCRIPTION
**Pull request by rfuest**
Thank you for helping out with embedded-graphics development! Please:

- [ ] Check that you've added passing tests and documentation
- [ ] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) if your changes affect the **public API**
- [ ] Run `rustfmt` on the project
- [ ] Run `just build` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

This PR makes drawing of 90° arcs consistent to drawing a quarter of a circle. Maybe this will help with #441. `Sector` will probably also need some tweaks, but that will be another PR.

### CODE REVIEW

**Comment by rfuest** [comment]
I've converted this to a draft, because arcs with even diameters aren't handled correctly. There is a test for this case, but it makes the same wrong assumptions than the actual code.

**Comment by rfuest** [comment]
It should now work as expected. The changes to the way the center point is calculated will also have some effect on `Sector` drawing, but I'll add additional tests for that in another dedicated PR.

File: src/primitives/arc/linear_equation.rs [hunk 1]
```diff
@@ -20,10 +20,17 @@ pub struct LinearEquation {
 impl LinearEquation {
     /// Create a new linear equation based on one point and one angle
     pub fn from_point_angle(point: Point, angle: Angle) -> Self {
-        let (a, b) = match angle.tan() {
-            None => (Real::from(1.0), Real::from(0.0)),
-            Some(a) => (-a, Real::from(-1.0)),
+        // FIXE: angle.tan() for 180.0 degrees isn't exactly 0 which causes problems when drawing
+        //       a single quadrant. Is there a better solution to fix this?
```
Comment (jamwaffles) [inline src/primitives/arc/linear_equation.rs:24]:
```suggestion
        // FIXME: angle.tan() for 180.0 degrees isn't exactly 0 which causes problems when drawing
        //        a single quadrant. Is there a better solution to fix this?
```

A stupid nit, but makes for more accurate grepping

**Review by jamwaffles — CHANGES_REQUESTED** [review]
I can't speak to the improvements on the algorithm, but the extra tests are a very welcome addition.

**Review by jamwaffles — APPROVED** [review]

**Comment by jamwaffles** [comment]
Please let me know if you decide to pick up #405, otherwise I'll take another look at it.

**Comment by rfuest** [comment]
> Please let me know if you decide to pick up #405, otherwise I'll take another look at it.

I haven't planed to, so give it a go if you like.

**Pull request merged**